docs/arch/adr-15.md

```diff
@@ -0,0 +1,40 @@
+# ADR 15: Drawing tools API
+
+June 19, 2019
+
+## Context
+
+The way the drawing tools currently function on Panoptes-Front-End (PFE) have numerous issues including:
+
+- Re-rendering the DOM more than necessary
+- Updating the classification annotation on each touch or click event
+
+Both of these can cause significant performance issues.
+
+- The separation concerns are not clear between components and stores. Multiple components can update what is being rendered making it hard to debug or add new features to. 
+
+## Decision
+
+What we do not want to do:
+- Re-render every time a click or touch event happens
+- Add to the annotations to the classification every time a click or touch event happens
+- Do not support more than one drawing task in a step
+- Use D3.js since it has its own internal data store and it would be complicated to integrate that with a observable stream.
+
+What we do want to do:
+- Have a component, the interactive layer, that manages the interaction with the marks and the click and touch events.
+  - It should not allow events to bubble so the events are encapsulated to just the interaction with the subject. This is to help prevent browser scroll during drawing.
```

Comment:
What does 'It' refer to here?

Reply:
Maybe refer to https://github.com/zooniverse/Panoptes-Front-End/pull/5411 which doesn't fix this but might be helpful to refer to.

Reply:
The interaction layer. I'll update that to state it explicitly.